src/qtum/qtumx86.cpp

```diff
@@ -371,6 +372,22 @@ uint32_t QtumHypervisor::ReadExternalStorage(uint32_t syscall, x86Lib::x86CPU& v
     return status;
 }
 
+uint32_t QtumHypervisor::SHA256(uint32_t syscall, x86Lib::x86CPU& vm) {
+    //eax = success
+    //ebx = original value
+    //ecx = sizeof original value
+    //edx = hash of original value
+    size_t len = vm.Reg32(ECX);
+    unsigned char* k = new unsigned char[len];
```

Comment:
It's technically possible to do this directly in the memory buffer to avoid a copy/allocation, but probably not very safe.. and stack allocation isn't safe for a user provided size